src/libYARP_dev/src/modules/ControlBoardWrapper/ControlBoardWrapper.cpp

```diff
@@ -939,7 +939,7 @@ void ControlBoardWrapper::run()
 //
 bool ControlBoardWrapper::setPid(int j, const Pid &p)
 {
-    int off=device.lut[j].offset;
+    int off; try{off = device.lut.at(j).offset;} catch(...){yError() << "joint number " << j <<  " out of bound [0-"<< controlledJoints << "] for part " << partName; return false; }
```

Comment:
@aerydna is there any specific reason why you make use of `try`+`catch` construct? Checking directly against the buffer overflow should be preferable in my opinion.